myndla-api/src/main/scala/no/ndla/myndlaapi/service/ArenaReadService.scala

```diff
@@ -482,6 +484,37 @@ trait ArenaReadService {
       } yield apiTopic
     }
 
+    def upvotePost(postId: Long, user: MyNDLAUser)(session: DBSession = AutoSession): Try[api.Post] = {
+      for {
+        maybePost     <- arenaRepository.getPost(postId)(session)
+        (post, owner) <- maybePost.toTry(NotFoundException(s"Could not find post with id $postId"))
+        upvoted       <- arenaRepository.getUpvoted(postId, user.id)(session)
+        _ <-
+          if (upvoted.isEmpty && owner.exists(_.id != user.id)) arenaRepository.upvotePost(postId, user.id)(session)
+          else Success(())
+        flags   <- arenaRepository.getFlagsForPost(postId)(session)
+        upvotes <- arenaRepository.getUpvotesForPost(postId)(session)
+        // The post is now upvoted if it wasn't initiated by the post's owner
+        newUpvotedState = upvoted.isEmpty && !owner.exists(_.id != user.id)
+        compiledPost    = CompiledPost(post, owner, flags, upvotes.length, newUpvotedState)
```

Comment:
Siden eiger ikkje kan upvote skal det være nok å sjekke på det.
```suggestion
        upvoted       <- arenaRepository.getUpvoted(postId, user.id)(session)
        notOwner      = owner.exists(_.id != user.id)
        _ <-
          if (upvoted.isEmpty && notOwner) arenaRepository.upvotePost(postId, user.id)(session)
          else Success(())
        flags   <- arenaRepository.getFlagsForPost(postId)(session)
        upvotes <- arenaRepository.getUpvotesForPost(postId)(session)
        // The post is now upvoted if it wasn't initiated by the post's owner
        compiledPost    = CompiledPost(post, owner, flags, upvotes.length, notOwner)
```

Reply:
Var alt forenklet noe lokalt, men kunne forenkle et lite hakk til med tanken herfra 🎉